Mathlib/AlgebraicGeometry/StructureSheaf.lean

```diff
@@ -0,0 +1,1224 @@
+  congr 1
+  change toStalk R p = _ ≫ (stalkIso R p).inv
+  erw [Iso.eq_comp_inv]
+  exact toStalk_comp_stalkToFiberRingHom R p
+#align algebraic_geometry.structure_sheaf.is_localization.to_stalk AlgebraicGeometry.StructureSheaf.IsLocalization.to_stalk
+
+instance openAlgebra (U : (Opens (PrimeSpectrum R))ᵒᵖ) : Algebra R ((structureSheaf R).val.obj U) :=
+  (toOpen R (unop U)).toAlgebra
+#align algebraic_geometry.structure_sheaf.open_algebra AlgebraicGeometry.StructureSheaf.openAlgebra
+
+@[simp]
+theorem openAlgebra_map (U : (Opens (PrimeSpectrum R))ᵒᵖ) (r : R) :
+    algebraMap R ((structureSheaf R).val.obj U) r = toOpen R (unop U) r :=
+  rfl
+#align algebraic_geometry.structure_sheaf.open_algebra_map AlgebraicGeometry.StructureSheaf.openAlgebra_map
+
+/-- Sections of the structure sheaf of Spec R on a basic open as localization of R -/
+instance IsLocalization.to_basicOpen (r : R) :
+    IsLocalization.Away r ((structureSheaf R).val.obj (op <| PrimeSpectrum.basicOpen r)) := by
+  convert (IsLocalization.isLocalization_iff_of_ringEquiv (S := Localization.Away r) _
+      (basicOpenIso R r).symm.commRingCatIsoToRingEquiv).mp
+      Localization.isLocalization
+  apply Algebra.algebra_ext
+  intro x
+  congr 1
+  exact (localization_toBasicOpen R r).symm
+#align algebraic_geometry.structure_sheaf.is_localization.to_basic_open AlgebraicGeometry.StructureSheaf.IsLocalization.to_basicOpen
+
+instance to_basicOpen_epi (r : R) : Epi (toOpen R (PrimeSpectrum.basicOpen r)) :=
+  ⟨fun {S} f g h => by
+    refine' IsLocalization.ringHom_ext (R := R)
+      (S := (structureSheaf R).val.obj (op <| PrimeSpectrum.basicOpen r)) _ _
+    exact h⟩
+#align algebraic_geometry.structure_sheaf.to_basic_open_epi AlgebraicGeometry.StructureSheaf.to_basicOpen_epi
+
+@[elementwise]
+theorem to_global_factors :
+    toOpen R ⊤ =
+      CommRingCat.ofHom (algebraMap R (Localization.Away (1 : R))) ≫
+        toBasicOpen R (1 : R) ≫
+        (structureSheaf R).1.map (eqToHom PrimeSpectrum.basicOpen_one.symm).op := by
+  rw [← Category.assoc]
+  change toOpen R ⊤ =
+    (CommRingCat.ofHom <| (toBasicOpen R 1).comp (algebraMap R (Localization.Away 1))) ≫
+    (structureSheaf R).1.map (eqToHom _).op
+  unfold CommRingCat.ofHom
+  rw [localization_toBasicOpen R, toOpen_res]
+#align algebraic_geometry.structure_sheaf.to_global_factors AlgebraicGeometry.StructureSheaf.to_global_factors
+
+instance isIso_to_global : IsIso (toOpen R ⊤) := by
+  let hom := CommRingCat.ofHom (algebraMap R (Localization.Away (1 : R)))
+  haveI : IsIso hom :=
+    IsIso.of_iso (IsLocalization.atOne R (Localization.Away (1 : R))).toRingEquiv.toCommRingCatIso
+  rw [to_global_factors R]
+  infer_instance
+#align algebraic_geometry.structure_sheaf.is_iso_to_global AlgebraicGeometry.StructureSheaf.isIso_to_global
+
+/-- The ring isomorphism between the ring `R` and the global sections `Γ(X, 𝒪ₓ)`. -/
+-- Porting note : was @[simps (config := { rhsMd := Tactic.Transparency.semireducible })]
```

Comment:
Are you clear about what is happening here? I'm not, so basically I'm saying "I'm reviewing your PR but this line I am not capable of reviewing properly".

Reply:
I am not sure what to do with this, but I see that `@[simps (config := { rhsMd := semireducible })]` in other ported file are just replaced with `simps!`